### PR TITLE
Test cases for facet integrals on quadrilaterals

### PR DESCRIPTION
**Pull request by miklos1**
Requires FFC updates to pass.
Upwind flux test cases on quadrilaterals are XFAILed.

### CODE REVIEW

File: tests/regression/test_poisson_mixed_strong_bcs.py [hunk 1]
```diff
@@ -27,13 +27,29 @@
...
-    DG = FunctionSpace(mesh, "DG", 0)
+    if quadrilateral:
+        S0 = FiniteElement("CG", "interval", 1)
+        S1 = FiniteElement("DG", "interval", 0)
+
+        T0 = FiniteElement("CG", "interval", 1)
+        T1 = FiniteElement("DG", "interval", 0)
+
+        DG_elt = OuterProductElement(S1, T1)
+        BDM_elt_h = HDiv(OuterProductElement(S1, T0))
```
Comment (dorugeber) [inline tests/regression/test_poisson_mixed_strong_bcs.py]: This looks like an RT element (strange choice of variable name!)

Comment (dorugeber) [inline tests/regression/test_poisson_mixed_strong_bcs.py]: You could make a "BDM-like" element by taking HDiv(CG1xDG1) + HDiv(DG1xCG1), though this isn't "the" BDM element on quads.  If this works, do it.  If it doesn't work, stick with the RT element and change some variable names.

Comment (miklos1) [inline tests/regression/test_poisson_mixed_strong_bcs.py]: Test passes with HDiv(CG1xDG1) + HDiv(DG1xCG1), I'm pushing the change soon.

**Comment by miklos1** [comment]
FFC updates have arrived.

**Pull request merged**